bundles/org.openhab.binding.iammeter/src/main/java/org/openhab/binding/iammeter/internal/IammeterHandler.java

```diff
@@ -0,0 +1,203 @@
+                for (IammeterWEM3080TChannel channelConfig : IammeterWEM3080TChannel.values()) {
+                    Channel channel = getThing().getChannel(channelConfig.getId());
+                    if (channel != null){
+                        State state = getDecimal(iammeterData.get(keyWord).getAsJsonArray().get(channelConfig.getRow())
+                                .getAsJsonArray().get(channelConfig.getCol()).toString());
+                        updateState(channel.getUID(), state);
+                    }
+                }
+            }
+            if (bRemoveChannels) {
+                if (!bExtraChannelRemoved) {
+                    thingStructureChanged(channelProfix);
+                }
+            }
+            updateStatus(ThingStatus.ONLINE);
+        // Very rudimentary Exception differentiation
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Communication error with the device: " + e.getMessage());
+        } catch (JsonSyntaxException je) {
+            logger.warn("Invalid JSON when refreshing source {}: {}", getThing().getUID(), je.getMessage());
+        } catch (Exception e) {
+            logger.warn("Error refreshing source {}: {}", getThing().getUID(), e.getMessage(), e);
+        }
+    }
+
+    protected void thingStructureChanged(String channelProfix) {
+        List<ChannelUID> noUsedItems = new ArrayList<>();
+        noUsedItems.add(new ChannelUID(channelProfix + ":frequency_a"));
+        noUsedItems.add(new ChannelUID(channelProfix + ":pf_a"));
+        noUsedItems.add(new ChannelUID(channelProfix + ":voltage_b"));
+        noUsedItems.add(new ChannelUID(channelProfix + ":current_b"));
+        noUsedItems.add(new ChannelUID(channelProfix + ":power_b"));
+        noUsedItems.add(new ChannelUID(channelProfix + ":importenergy_b"));
+        noUsedItems.add(new ChannelUID(channelProfix + ":exportgrid_b"));
+        noUsedItems.add(new ChannelUID(channelProfix + ":frequency_b"));
+        noUsedItems.add(new ChannelUID(channelProfix + ":pf_b"));
+        noUsedItems.add(new ChannelUID(channelProfix + ":voltage_c"));
+        noUsedItems.add(new ChannelUID(channelProfix + ":current_c"));
+        noUsedItems.add(new ChannelUID(channelProfix + ":power_c"));
+        noUsedItems.add(new ChannelUID(channelProfix + ":importenergy_c"));
+        noUsedItems.add(new ChannelUID(channelProfix + ":exportgrid_c"));
+        noUsedItems.add(new ChannelUID(channelProfix + ":frequency_c"));
+        noUsedItems.add(new ChannelUID(channelProfix + ":pf_c"));
+        ThingBuilder thingBuilder = editThing();
+        for (ChannelUID chl : noUsedItems) {
+            thingBuilder.withoutChannel(chl);
+        }
+        updateThing(thingBuilder.build());
+        bExtraChannelRemoved = true;
+    }
+
+    private State getDecimal(String value) {
+        try {
+            return new DecimalType(new BigDecimal(value));
```

Comment:
You need to use `QuantityType` to make use of Units of Measure. You could store the unit type in `IammeterWEM3080Channel`.
```suggestion
            return QuantityType.valueOf(Float.parseFloat(value), SmartHomeUnits.VOLT);
```

Reply:
code changed.